src/components/ha-navigation-picker.ts

```diff
@@ -0,0 +1,225 @@
+import { ComboBoxLitRenderer } from "@vaadin/combo-box/lit";
+import { css, html, LitElement, PropertyValues, TemplateResult } from "lit";
+import { customElement, property, query, state } from "lit/decorators";
+import { fireEvent } from "../common/dom/fire_event";
+import { titleCase } from "../common/string/title-case";
+import {
+  fetchConfig,
+  fetchDashboards,
+  LovelaceViewConfig,
+} from "../data/lovelace";
+import { PolymerChangedEvent } from "../polymer-types";
+import { HomeAssistant } from "../types";
+import "./ha-combo-box";
+import type { HaComboBox } from "./ha-combo-box";
+import "./ha-icon";
+
+type NavigationItem = {
+  path: string;
+  icon: string;
+  title: string;
+};
+
+const DEFAULT_ITEMS = [
```

Comment:
Instead of this, why not use hass.panels?

Reply:
We can replace the `fetchDashboards` by `hass.panels`. So we just have to fetch the config for each panel starting with `lovelace`. Seems a good idea 🙂